server/routes/avatar/utils.js

```diff
@@ -0,0 +1,99 @@
+import sharp from 'sharp';
+import { throttle } from 'underscore';
+
+import { Cookies } from 'meteor/ostrio:cookies';
+
+import { Users } from '../../../app/models/server';
+import { getAvatarColor } from '../../../app/utils';
+import { settings } from '../../../app/settings/server';
+
+const FALLBACK_LAST_MODIFIED = 'Thu, 01 Jan 2015 00:00:00 GMT';
+
+const cookie = new Cookies();
+
+export const getCacheTime = (req) => req.query.cacheTime || settings.get('Accounts_AvatarCacheTime');
+
+export const serveAvatar = (avatar, req, res) => {
+	res.setHeader('Last-Modified', FALLBACK_LAST_MODIFIED);
+
+	if (['png', 'jpg', 'jpeg'].includes(req.query.format)) {
+		res.setHeader('Content-Type', `image/${ req.query.format }`);
+		sharp(new Buffer(avatar))
+			.toFormat(req.query.format)
+			.pipe(res);
+		return;
+	}
+	res.setHeader('Content-Type', 'image/svg+xml');
+
+	res.write(avatar);
+	res.end();
+};
+
+export const wasFallbackModified = (reqModifiedHeader, res) => {
+	if (!reqModifiedHeader || reqModifiedHeader !== FALLBACK_LAST_MODIFIED) {
+		return true;
+	}
+	res.writeHead(304);
+	res.end();
+	return false;
+};
+
+function isUserAuthenticated(req) {
+	const headers = req.headers || {};
+	const query = req.query || {};
+
+	let { rc_uid, rc_token } = query;
+
+	if (!rc_uid && headers.cookie) {
+		rc_uid = cookie.get('rc_uid', headers.cookie) ;
+		rc_token = cookie.get('rc_token', headers.cookie);
+	}
+
+	if (!rc_uid || !rc_token || !Users.findOneByIdAndLoginToken(rc_uid, rc_token)) {
```

Comment:
Here we can simply return because the result is already Boolean, don't need this `if else`.